### PR TITLE
fix dateTime's type

### PR DESCRIPTION
**Pull request by sowu880**

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Utility/DateTimeUtilityTests.cs [hunk 1]
```diff
@@ -45,16 +46,16 @@ public static IEnumerable<object[]> GetDateDataForDateShiftWithPrefix()
...
         public static IEnumerable<object[]> GetDateDataForDateShiftButShouldBeRedacted()
         {
-            yield return new object[] { new Date("2015-02"), new Date("2015") };
+            yield return new object[] { new Date("2015-02"), PartialDateTime.Parse("2015") };
             yield return new object[] { new Date("1925-02-07"), null };
         }
 
         public static IEnumerable<object[]> GetDateTimeDataForRedact()
         {
-            yield return new object[] { new FhirDateTime("2015"), new FhirDateTime("2015") };
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Utility/DateTimeUtilityTests.cs:54]: We should add test cases for instant types too

File: src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs [hunk 2]
```diff
@@ -42,7 +43,14 @@ public static ProcessResult RedactDateNode(ElementNode node, bool enablePartialD
...
                 {
                     string yearOfDate = matchedGroups[s_yearIndex].Value;
-                    node.Value = IndicateAgeOverThreshold(matchedGroups) ? null : yearOfDate;
+                    if (IndicateAgeOverThreshold(matchedGroups))
+                    {
+                        node.Value = null;
+                    }
+                    else
+                    {
+                        node.Value = PartialDateTime.Parse(yearOfDate);
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs:52]: It seems there's another type named FhirDateTime. Could we use FhirDateTime here?

Comment (sowu880) [reply src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs:52]: FhirDateTime will be converted to  PartialDateTime in fhir.

Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs:52]: Got it. Is PartialDateTime the default type of node.Value for dates, dateTime and instant? Will we change the object type here?

Comment (sowu880) [reply src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs:52]: Yes, dates, dateTime and instant will be transformed into PartialDateTime.

File: src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Utility/DateTimeUtilityTests.cs (hunk 1 above)
Comment (sowu880) [reply src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Utility/DateTimeUtilityTests.cs:54]: Added.

**Pull request closed without merging**